### PR TITLE
feat(ZNTA-747) add RTL support to alpha editor

### PR DESCRIPTION
**Pull request by kathryngo**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-747

# Add css that turns the text in the alpha interface from LTR to RTL.

## Reasoning for using manual css
I tried some webpack plugins but we do not want entire stylesheets flipped as this really messed with the UX. We only want the text and panels in the editor flipped, and text in the modals and inputs. (If I am missing something please comment).

Instead a 'rtl' class has been added manually to components where required to get the text displaying right-to-left without complicating the layout. A 'ltr' class has been created also. For the purpose of clarity, I have set the source language as LTR and the target as RTL but of course this will need to be changed so the language used is based on detection.

## RTL css class locations
These are located in the following editor files:
```editor/components/GlossarySearchInput/index.js```
```editor/containers/NavHeader.js```
```editor/containers/TranslationInfoPanel/index.js```
```editor/components/TransUnitTranslationPanel/index.js``` 
```editor/components/GlossaryTerm/index.js```
```editor/components/GlossaryTermModal/component.js```
```editor/containers/SuggestionDetailsModal/PlainSuggestionContents.js```
```editor/containers/SettingsPanel.js```
```editor/components/SuggestionTranslations.js```

## Suggestions for implementation
I would suggest writing something that applies this class to these modules when RTL languages are used. ie Language detection - see Jira issue comments for details.

### Backend work will need to be completed first.
* Also you will notice that a lot of yarn dependencies have been removed. That is because I removed the package react-styleguidist which was not being used

## PREVIEW
rtl-current
gloss-modal

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/583)

### CODE REVIEW

File: server/zanata-frontend/src/frontend/package.json [hunk 1]
```diff
@@ -138,6 +137,7 @@
     "redux-sac": "0.5.3",
     "redux-thunk": "1.0.3",
     "redux-watch": "1.1.1",
+    "rtl-detect": "^1.0.0",
```
Comment (kathryngo) [inline server/zanata-frontend/src/frontend/package.json]: maybe someone more skilled than I can get this working?

Comment (kathryngo) [reply server/zanata-frontend/src/frontend/package.json]: @aeng ^ re: our discussion today - will leave this PR so someone else can try and get the language detection working

**Comment by kathryngo** [comment]
Am removing the text direction functionality from the search toolbar since either target or source text can be searched for - best to keep LTR.

**Comment by huangp** [comment]
:lgtm: -(Awaiting review) +Reviewed 

---

Reviewed 1 of 8 files at r2, 3 of 20 files at r4, 24 of 26 files at r6, 1 of 1 files at r7, 5 of 5 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/583#-:-KxMAMci8PAO04zwrE1p:b-ew2adl)*

**Comment by aeng** [comment]
Reviewed 1 of 8 files at r2, 4 of 20 files at r4, 24 of 26 files at r6, 1 of 1 files at r7, 5 of 5 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/583)*

**Comment by kathryngo** [comment]
How should we go about testing this PR with users?

**Comment by aeng** [comment]
Unless they run zanata locally, theres no way for them to test it. The best we can do is deploy it to staging that will be available internally.

**Comment by spathare21** [comment]
Suggestions details modal not showing on version/doc name click, also no glossary result shown. translation shows RTL for rtl languages.

**Comment by aeng** [comment]
@spathare21 

> Suggestions details modal not showing on version/doc name click
Was the modal showing the information before this PR?

> also no glossary result shown. 
Is there a translation in the glossary for the string? Make sure you reindex too. Anyhow, this should be separate issue

> translation shows RTL for rtl languages.
Good!

**Comment by spathare21** [comment]
@aeng yes the suggestion details panel show's before this PR after clicking on project version/doc from suggestions panel.

**Comment by spathare21** [comment]
verified LTR and RTL works fine also suggestion details shown up.

**Pull request merged**